app/src/main/java/live/hms/app2/ui/meeting/MeetingViewModel.kt

```diff
@@ -1370,8 +1370,12 @@ class MeetingViewModel(
         hmsSDK.startHLSStreaming(config, object : HMSActionResultListener {
             override fun onError(error: HMSException) {
                 viewModelScope.launch {
-                    _events.emit(Event.Hls.HlsError(error))
-                    hlsToggleUpdateData.postValue(false)
+                    if (error.message.contains("invalid input")) {
```

Comment:
Should never do string matching of errors. Only match on error codes if available.
But why are you starting hls after invalid input?

Reply:
actually its an error that is thrown when `destinations` are null , here is the web pr which i got logic from 

https://github.com/100mslive/100ms-web/blob/main/src/components/Streaming/HLSStreaming.jsx#L56